code/game/objects/items/tools/flame_tools.dm

```diff
@@ -123,6 +123,12 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 
 	attack_verb = list("burnt", "singed")
 
+/obj/item/tool/match/afterattack(atom/target, mob/living/carbon/human/user, proximity_flag, click_parameters)
+	if(istype(user) && istype(target, /obj/item/clothing/shoes/marine) && user.shoes == target && light_match())
+		user.visible_message(SPAN_NOTICE("<b>[user]</b> strikes \the [src] against their [target.name], igniting it!"), SPAN_NOTICE("You strike \the [src] against your [target.name], igniting it!"), max_distance = 3)
+		return
+	return ..()
+
```

Comment:
imo this might trivialise taking the actual matchbox around with you to light them

give them a chance of snapping in half and breaking on your boot